utils/streams.js

```diff
@@ -0,0 +1,129 @@
+
+
+function reverse(str) {
+    str
+        .pipe(through(function (buffer, encodding, done) {
+            this.push(util.reverse(buffer));
+            done();
+            process.exit(0);
+        }))
+        .pipe(process.stdout);
+}
+
+function transform(str) {
+    str
+        .pipe(through(function (buffer, encodding, done) {
+            this.push(util.toUpperCase(buffer));
+            done();
+            process.exit(0);
+        }))
+        .pipe(process.stdout);
+}
+
+function outputFile(filePath) {
+    if (!filePath) program.help();
+    fs.createReadStream(filePath)
+        .on('error', function () {
+            console.error('File not found: ' + filePath);
+            program.help();
+        })
+        .pipe(process.stdout);
+}
+
+function convertFromFile(filePath) {
+    fs.createReadStream(filePath)
+        .on('error', function () {
+            console.error('File not found: ' + filePath);
+            program.help();
+        })
+        .pipe(csvConverter)
+        .pipe(process.stdout);
+}
+
+function convertToFile(filePath) {
+    fs.createReadStream(filePath)
+        .on('error', function () {
+            console.error('File not found: ' + filePath);
+            program.help();
+        })
+        .pipe(csvConverter)
+        .pipe(fs.createWriteStream(filePath.substring(0, filePath.lastIndexOf('.')) + '.json'));
+}
+
+function cssBundler(filePath) {
+    console.log(filePath);
+    readdir(filePath)
+        .then(files => {
+            files.forEach(fileName => {
+                appendToResult(filePath + '/' + fileName);
+            });
+            appendToResult(appendixPath);
+        })
+        .catch(console.error);
+}
+
+function appendToResult(filePath) {
+    readFile(filePath)
+        .then(content => {
+            appendFile(resultPath, content);
+        });
+}
+
+function call(action, value) {
+    if (!value) program.help();
+    action(value);
+}
+
+function start() {
+    program
+        .name('module3')
+        .option('-a, --action <action>', 'action to perform')
+        .option('-f, --file <filePath>', 'path to the existing file')
+        .option('-p, --path <filePath>', 'path to the existing dir')
+        .on('--help', function () {
+            console.log('Available actions:');
+            console.log('reverse          - reverse string data');
+            console.log('transform        - transform string data to uppercase');
+            console.log('outputFile       - print the contents of file specified in --file option');
+            console.log('convertFromFile  - print the contents of .csv file specified in --file option as json');
+            console.log('convertToFile    - convert .csv file specified in --file option to .json');
+            console.log('cssBundler       - merge .css files specified in --path directory');
+        })
+        .parse(process.argv);
+
+    if (program.action === 'reverse') {
+        reverse(process.stdin);
```

Comment:
Instead of if/else in this case is better to use
```javascript
const actionList = {};
actionList.action1 = () => {};
actionList.action2 = () => {};
...

program.action = 'action1';
actionList[program.action](process.stdin);
call(actionList[program.action], program.file);
```